src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -174,8 +174,11 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
             return data;
         }
         for (FilePath test : tests) {
-            Matcher m = TEST.matcher(test.getRemote().replace('\\', '/'));
-            assert m.matches();
```

Comment:
matches must be called for group information to be populated but when assertions are disabled this doesn't get invoked